runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -84,7 +84,7 @@ public WorkFlowExecutionController(EventsDAO eventsDAO, StateMachinesDAO stateMa
 
         final List<String> triggeredEvents = eventsDAO.findTriggeredEventsNamesBySMId(stateMachine.getId());
         Set<State> initialStates = context.getInitialStates(new HashSet<>(triggeredEvents));
-        executeStates(stateMachine.getId(), initialStates);
+        executeStates(stateMachine.getId(), stateMachine.getName(), initialStates);
```

Comment:
Instead of sending name as one more parameter, directly pass state machine object itself, and take the required params there.